use-cases/NIC-passthrough-and-Kata.md

```diff
@@ -0,0 +1,217 @@
+
+Say I want to pass **enp0s20f1** to the container (see what interfaces you have by running `$ ip link` ). 
+
+Get address: 
+
+```bash
+$ NIC=<interface>
+```
+For example: 
+```bash
+$ NIC=enp0s20f1
+```
+Place in BDF:
+```bash
+$ BDF=$(sudo lshw -class network -businfo -numeric | grep ${NIC} | awk '{print $1;}' | cut -d@ -f2)
+```
+Test BDF:
+```bash
+$ echo $BDF
+```
+
+```
+Expected results (something like):
+
+0000:00:14.1
+```
+
+#### Load *vfio-pci*  kernel module
+
+```bash
+$ sudo modprobe -i vfio-pci
+```
+Test that the module loaded correctly: 
+
+```bash
+$ sudo lsmod | grep vfio
+```
+
+#### Unbind the Interface from its current driver. 
+
+```bash
+$ sudo echo $BDF | sudo tee /sys/bus/pci/devices/$BDF/driver/unbind
+```
+
+#### Find vendor & device ID
+
+```bash
+$ sudo lspci -n -s $BDF
+```
+
+```
+Expected results (something like):
+
+00:14.1 0200: 8086:1f40 (rev 03)
+```
+
+
+#### Bind the device to vfio-pci using the result of the previous step.
+
+```bash
+$ sudo echo 8086 1f40 | sudo tee /sys/bus/pci/drivers/vfio-pci/new_id
+```
+
+> **Note:** The ` xxxx xxxx ` format is of the correct form and not a mistake.
+
+#### Check what the new device name is, now under vfio
+
+```bash
+$ sudo ls /dev/vfio
+```
+```
+Expected results (something like):
+
+ 39  vfio
+```
+
+> **Note:** The requested outcome here is the newly created “39”.
+
+#### All set here. Time to move to the Container’s side of the deal.
+
+
+### Part 2 – The Container
+
+#### Create the container you wish to pass the NIC into
+
+```bash
+docker run -it --runtime=kata-runtime --name <choose a name> --device /dev/vfio/<device name from previous step> -v /dev:/dev <image> <command>
+```
+   For example:
+
+```bash
+docker run -it --runtime=kata-runtime --name vfio_con --device /dev/vfio/39 -v /dev:/dev ubuntu bash
+```
+If you aren’t sure what just happened back there, here’s a breakdown of the command:
+- `docker run` - creates and runs the container. 
+- ` -it ` - docker command flags for “interactive” and “tty”.
+- ` --runtime=kata-runtime ` - tells docker to run the container as a KC (if you set kata-runtime as the default runtime, this option is redundant).
+- ` --name vfio_con ` – name the container. It can be any name you choose. 
+- ` --device /dev/vfio/39 ` – this one is the important part. With this command, you tell docker to pass the NIC to the container. 
+- `-v /dev:/dev `– here you tell docker to map the devices folder of the host to the devices folder of the container.
+- ` ubuntu `– choose the container image you want to run. 
+- ` bash ` - the command you what the container to run. 
+
+ 
+
+#### Find forwarded device
+Update and install pciutils (Ubuntu)
+
+```bash
+$ sudo apt update ; apt install pciutils
+```
+
+List devices:
+```bash
+$ sudo lspci –nn -D
+```
+
+> **Note:** The [XXXX:XXXX] address you are looking for is the same one you used in the previous steps.
+
+#### Bind the passed device with a driver in the container.
+/ TODO
```

Comment:
Are you planning to update these?

Reply:
No, Please see: [603105747](https://github.com/kata-containers/runtime/issues/1876#issuecomment-603105747). Thanks